### PR TITLE
gvls: init at 0.20

### LINKED ISSUE #83118: Package gvls
gnome-builder now needs gvls for vala support: https://gitlab.gnome.org/esodan/gvls

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Fixes https://github.com/NixOS/nixpkgs/issues/83118. 
It's kinda weird, but this project puts a lot of things in `Requires` in their pkgconfig files.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/gvls/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    jsonrpc-glib
+    libgee
+    vala
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Language Server Provider for Vala";
+    homepage = "https://gitlab.gnome.org/esodan/gvls";
+    license = licenses.lgpl3;
+    maintainers = with maintainers; [ worldofpeace ];
```
Comment (worldofpeace) [inline pkgs/development/tools/gvls/default.nix]: hmm, maybe this should be the gnome team

File: pkgs/development/tools/gvls/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+, pango
+, gtksourceview3
+, libgee
+, gobject-introspection
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gvls";
+  version = "0.14.3";
```
Comment (doronbehar) [inline pkgs/development/tools/gvls/default.nix:20]: Should be updated to 0.16.0: https://gitlab.gnome.org/esodan/gvls/-/tags

Comment (worldofpeace) [reply pkgs/development/tools/gvls/default.nix:20]: Yep, was planned.

File: pkgs/development/tools/gvls/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    json-glib
+    jsonrpc-glib
+    libgee
+    vala
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Language Server Provider for Vala";
+    homepage = "https://gitlab.gnome.org/esodan/gvls";
+    license = licenses.lgpl3;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gvls/default.nix:58]: Plus or only?

File: pkgs/development/tools/gvls/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    glib
+    gtk3
+    gtksourceview3
+    json-glib
+    jsonrpc-glib
+    libgee
+    vala
+  ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gvls/default.nix:55]:
```suggestion
  meta = with lib; {
```

**Review by jtojnar — APPROVED** [review]

**Comment by jtojnar** [comment]
There has been a version 20 released.

**Comment by doronbehar** [comment]
Updated to 0.20 and changed the patch to be fetched from a PR upstream: https://gitlab.gnome.org/esodan/gvls/-/merge_requests/14

Will merge if no one will object tomorrow or later.

**Pull request closed without merging**